x-pack/plugins/cases/server/services/attachments/index.ts

```diff
@@ -283,34 +281,29 @@ export class AttachmentService {
       this.log.debug(`Attempting to POST a new comment`);
 
       const soExtractor = getAttachmentSOExtractor(attributes);
-
-      const { transformedFields: migratedAttributes, references: refsWithExternalRefId } =
-        soExtractor.extractFieldsToReferences<AttachmentAttributesWithoutSORefs>({
-          data: attributes,
-          existingReferences: references,
-        });
-
       const { attributes: extractedAttributes, references: extractedReferences } =
-        extractPersistableStateReferencesFromSO(migratedAttributes, {
-          persistableStateAttachmentTypeRegistry: this.persistableStateAttachmentTypeRegistry,
-        });
+        extractAttachmentSOReferences(
```

Comment:
Same here I think we can move `getAttachmentSOExtractor` into the `extract*` helper.